src/main/java/com/google/step/servlets/UserPostUrlServlet.java

```diff
@@ -0,0 +1,26 @@
+package com.google.step.servlets;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that handles usr upload url. */
+@WebServlet("/api/user-post-url/*")
+public class UserPostUrlServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    long id;
+    try {
+      String idString = request.getPathInfo().substring(1);
+      id = Long.parseLong(idString);
+    } catch (NumberFormatException | IndexOutOfBoundsException e) {
+      response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+      return;
+    }
+    response.setContentType("text/html");
+    response.getWriter().println(ImageUploader.getUploadUrl("/api/users/" + String.valueOf(id)));
```

Comment:
```suggestion
    response.getWriter().println(ImageUploader.getUploadUrl("/api/users/" + id));
```

I'm note sure which is better practice, you can keep your way if you want